### DISCUSSION TITLE
Using livewire sortable with Spatie Media Library

### DISCUSSION THREAD

**Discussion by Sindoweb** [livewire/livewire#3227] [1. Help] [answered]
Hey all. I have an issue using livewire sortable with the Spatie Media Library.

Situation:
A user can upload images to a project in a CMS system. These images are sortable by drag and drop.

Problem:
Everything works as expected. Images can be dragged and dropped - the new order is saved into the database through a livewire function. So far so good.

The issue I'm having is that the reload of the image list through livewire shows the situation before the sort - the images flip back in their original order (although the database fields are updated correctly). Once I reload the entire page the sorting is correct and the new image list is shown as it should.

Livewire code:

```
namespace App\Http\Livewire;

use App\Models\Media;
use Livewire\Component;

class ProjectsDetailImages extends Component
{
    public $item;
    public $images;

    public function render()
    {
        return view('livewire.projects-detail-images');
    }

    public function updateImagesOrder($list)
    {
        foreach($list as $item):
            $media = Media::find($item['value']);
            $media->order_column = $item['order'];
            $media->save();
        endforeach;

        $this->images = $this->item->getMedia('images');
    }
}
```

Livewire blade:

```
<div>
    @if( !empty($images) )
            <div class="grid grid-cols-4 gap-4" wire:sortable="updateImagesOrder">
                @foreach($images as $image)
                    <div class="relative" wire:sortable.item="{{ $image->id }}" wire:key="item-{{ $image->id }}">
                        <img src="{{ $image->getUrl() }}" alt="" class="w-full">
                    </div>
                @endforeach
            </div>
    @endif
</div>
```

Any thoughts would be appreciated.

Kind regards,
Dominique

**Comment by drbornot** [comment, marked as answer]
maybe you need refresh the item after store the changes like:
```
$this->item->refresh();
$this->images = $this->item->getMedia('images');
```

**Reply by Sindoweb** [reply]
That worked!

For some reason I did $this->images->refresh(); which resulted in an error.

Thanks!
